### PR TITLE
Hard encoded esr trx inherited

### PR DESCRIPTION
**Pull request by metas-rc**

### CODE REVIEW

File: backend/de.metas.adempiere.adempiere/base/src/main/java-legacy/org/adempiere/ad/dao/impl/TypedSqlQuery.java [hunk 1]
```diff
@@ -514,6 +514,8 @@ else if (postQueryFilter.accept(model))
 		finally
 		{
 			DB.close(rs, pstmt);
+			rs = null;
+			pstmt = null;
```
Comment (teosarca) [inline backend/de.metas.adempiere.adempiere/base/src/main/java-legacy/org/adempiere/ad/dao/impl/TypedSqlQuery.java:518]: futile, but let it there now.

File: backend/de.metas.payment.esr/src/main/java/de/metas/payment/esr/processor/impl/LoadESRImportFileWorkpackageProcessor.java [hunk 2]
```diff
@@ -25,22 +25,22 @@
...
+import java.util.List;
+
 /**
  * Import the esr from the file which is stored in attachment
  *
  * @author cg
  *
  */
-public class LoadESRImportFileWorkpackageProcessor implements IWorkpackageProcessor
+public class LoadESRImportFileWorkpackageProcessor extends WorkpackageProcessorAdapter
```
Comment (teosarca) [inline backend/de.metas.payment.esr/src/main/java/de/metas/payment/esr/processor/impl/LoadESRImportFileWorkpackageProcessor.java:43]: i think u have to override isRunInTransaction() and return false

File: backend/de.metas.payment.esr/src/main/java/de/metas/payment/esr/api/impl/ESRImportBL.java [hunk 3]
```diff
@@ -147,6 +147,7 @@ private final void lockAndProcess(
 			@NonNull final I_ESR_Import esrImport,
 			@NonNull final Runnable processor)
 	{
+		trxManager.assertThreadInheritedTrxNotExists();
```
Comment (teosarca) [inline backend/de.metas.payment.esr/src/main/java/de/metas/payment/esr/api/impl/ESRImportBL.java:150]: there is also a process which calls this.
make sure that process runs out of trx too.

File: backend/de.metas.adempiere.adempiere/base/src/main/java-legacy/org/adempiere/ad/dao/impl/TypedSqlQuery.java (hunk 1 above)
Comment (metas-rc) [reply backend/de.metas.adempiere.adempiere/base/src/main/java-legacy/org/adempiere/ad/dao/impl/TypedSqlQuery.java:518]: It's the first thing I tried  when I saw all the other methods had this. But I saw it's irrelevant, Taking it out

File: backend/de.metas.payment.esr/src/main/java/de/metas/payment/esr/process/ESR_Complete_Process.java [hunk 4]
```diff
@@ -36,7 +36,13 @@ protected String doIt() throws Exception
 			throw new FillMandatoryException(I_ESR_Import.COLUMNNAME_ESR_Import_ID);
 		}
 
-		final I_ESR_Import esrImport = InterfaceWrapperHelper.create(p_ESR_Import_ID, I_ESR_Import.class);
+		final I_ESR_Import esrImport = InterfaceWrapperHelper.create(getCtx(), p_ESR_Import_ID, I_ESR_Import.class, get_TrxName());
```
Comment (teosarca) [inline backend/de.metas.payment.esr/src/main/java/de/metas/payment/esr/process/ESR_Complete_Process.java:39]: @metas-rc  hmm, what's the reason behind this?

Why we cannot simply use TRXNAME_ThreadInerited?

**Review by teosarca — APPROVED** [review]

**Pull request merged**